snitun/server/listener_sni.py

```diff
@@ -121,22 +122,39 @@ async def _proxy_peer(
             _LOGGER.error("Can't read source IP")
             return
 
+        pause_future: asyncio.Future[None] | None = None
+
+        def pause_resume_reader_callback(pause: bool) -> None:
+            """Pause and resume reader."""
+            nonlocal pause_future
+            if pause:
+                pause_future = self._loop.create_future()
+            elif pause_future:
+                pause_future.set_result(None)
+                pause_future = None
+
         # Open multiplexer channel
         try:
-            channel = await multiplexer.create_channel(ip_address)
+            channel = await multiplexer.create_channel(
+                ip_address,
+                pause_resume_reader_callback,
+            )
         except MultiplexerTransportError:
             _LOGGER.error("New transport channel to peer fails")
             return
 
-        from_proxy = None
+        from_proxy: asyncio.Future[None] | asyncio.Task[bytes] | None = None
         from_peer = None
         try:
             await channel.write(client_hello)
 
             # Process stream into multiplexer
             while not transport.is_closing():
                 if not from_proxy:
-                    from_proxy = self._loop.create_task(reader.read(4096))
+                    if pause_future:
+                        from_proxy = cast(asyncio.Future[None], pause_future)
```

Comment:
cast because of the nonlocal above, should refactor to get rid of this

Reply:
Need to make sure we are pausing the right reader.  It should be the one that we are reading data from that will be delivered to the remote side input